app/models/auth_token.rb

```diff
@@ -33,12 +35,28 @@ def self.create_multiple(array_of_tokens)
     end
   end
 
-  def high_rate_limit?(api_version)
-    return v4_remaining_rate > 500 if api_version == :v4
+  def fetch_resource_limits
+    client = github_client
+
+    v3_stat = client.rate_limit!.remaining
+    v4_stat = client.last_response.data.resources.graphql.remaining
 
-    github_client.rate_limit.remaining > 500
+    {
+      v3: v3_stat,
+      v4: v4_stat,
+    }
+  end
+
+  def safe_to_use?(api_version)
+    resource_limits = fetch_resource_limits
+    remaining = resource_limits.fetch(api_version)
+
+    remaining > LOW_RATE_LIMIT_REMAINING_THRESHOLD
   rescue Octokit::Unauthorized, Octokit::AccountSuspended
     false
+  rescue StandardError
+    StructuredLog.capture("FAILED_READING_GITHUB_RATE_LIMITS", { api_version: api_version, auth_token_id: id, remaining: remaining })
+    false
```

Comment:
It might be more appropriate to just let the exception raise at this point, but I do want to see when/how often it actually comes up